### PR TITLE
Android syncv2 setup infobar

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves  https://github.com/brave/brave-browser/issues/10822

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/10822) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
I. Do not show informer on clean install
1. Install this version 
2. Launch 
3. Ensure there is no infobar

II. Informer must be shown after update from sync v1 
1. Install old version with sync v1, like 1.11.96
2. Enable sync v1
3. Update with this PR version (some abobe 1.13.51)
4. Launch browser
5. Ensure informer is shown

III. Informer must not be shown when it was shown once
1. Follow steps (II.)
2. Close browser
3. Launch browser - informer must not be shown

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
Here is how it looks 

Screenshot_20200724-191410

File: components/brave_sync/brave_sync_prefs.h [hunk 1]
```diff
@@ -25,6 +25,9 @@ class PrefRegistrySyncable;
 
 namespace brave_sync {
 
+extern const char kSyncV2MigrateNoticeDismissed[];
```
Comment (darkdh) [inline components/brave_sync/brave_sync_prefs.h:28]: Can't we use JNI to access these two prefs? They are all hidden intentionally and can only be accessed by Prefs methods.

Comment (AlexeyBarabash) [reply components/brave_sync/brave_sync_prefs.h:28]: Yes, we can,  if this is vital.

Thanks for the notice, will update.

**Comment by AlexeyBarabash** [comment]
CI succeeded, but there are conflicts and one notice to fix.

File: components/brave_sync/brave_sync_prefs.h (hunk 1 above)
Comment (AlexeyBarabash) [reply components/brave_sync/brave_sync_prefs.h]: @darkdh 
fixed at https://github.com/brave/brave-core/pull/6204/commits/dc9d77acf84584be876acc25641cab787139eaf5

**Comment by AlexeyBarabash** [comment]
CI failed for `test-browser` on Linux only (https://ci.brave.com/job/pr-brave-browser-android_syncv2_setup_infobar/2/execution/node/597/log/).
```
18:44:18  [31432:31432:0728/154418.452621:WARNING:CONSOLE(11)] "Could not find style data in module named br-shared-style", source: chrome://resources/polymer/v1_0/polymer/polymer-extracted.js (11)
18:44:18  ../../brave/components/brave_rewards/browser/test/rewards_browsertest.cc:276: Failure
18:44:18  Value of: base::StartsWith( current_url.spec(), braveledger_uphold::GetUrl() + "/signup/step2", base::CompareCase::INSENSITIVE_ASCII)
18:44:18    Actual: false
18:44:18  Expected: true
18:44:18  Stack trace:
18:44:18  #0 0x560e5eee7f59 (/home/ubuntu/jenkins/workspace/pr-brave-browser-android_syncv2_setup_infobar/src/out/Release/brave_browser_tests+0x5c40f58)
18:44:18  
18:44:18  [  FAILED  ] RewardsBrowserTest.NotVerifiedWallet, where TypeParam =  and GetParam() =  (1575 ms)
18:44:18  [604/605] RewardsBrowserTest.NotVerifiedWallet (1830 ms)
18:44:20  [605/605] RewardsContributionBrowserTest.TipUnverifiedPublisher (1830 ms)
18:44:20  1 test failed:
18:44:20      RewardsBrowserTest.NotVerifiedWallet (../../brave/components/brave_rewards/browser/test/rewards_browsertest.cc:241)
18:44:20  null
18:44:20  null
```
it is not supposed to be touched by this PR.
Excluding all except Android and re-launching PCI

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Android 10 running 1.14.10 x64 nightly build
- Verified installing 1.12.21 and upgrading to 1.14.10 shows the upgrade message
   

- Verified clicking on `Begin Setup` opens up sync settings to optin to new sync v2
- Verified closing and restarting after the info bar is shown doesn't show again

**Pull request merged**